packages/@icon-magic/config-reader/src/config.ts

```diff
@@ -0,0 +1,102 @@
+import * as path from 'path';
+import * as glob from 'glob';
+import { loadConfigFile } from './config-loader';
+import { validateConfigSchema } from './config-validator';
+import { Icon } from '@icon-magic/icon-models';
+import { IconSetHashInterface } from '@icon-magic/icon-models/src/interface';
+import { isDirectory } from '@icon-magic/icon-models/dist/src/utils/files';
+
+export class Config {
+  iconSetHash: IconSetHashInterface;
+
+  constructor(configFiles: string[]) {
+    this.iconSetHash = new Map();
+    for (let configFile of configFiles) {
+      //validate all the config files
+      const configJson = loadConfigFile(configFile);
```

Comment:
Feel free to disregard, but should validation happen in the config loader?

Reply:
Validation is happening within the config-reader package, technically. I wish I could think of a better name for this package without having to use the word "manager"